config/spotbugs-exclude.xml

```diff
@@ -104,6 +104,8 @@
       <Class name="com.puppycrawl.tools.checkstyle.checks.coding.MatchXpathCheck"/>
       <Class name="com.puppycrawl.tools.checkstyle.checks.OuterTypeFilenameCheck" />
       <Class name="com.puppycrawl.tools.checkstyle.checks.regexp.RegexpCheck" />
+      <Class name=
+             "com.puppycrawl.tools.checkstyle.checks.javadoc.JavadocLeadingAsteriskAlignCheck" />
```

Comment:
Added this suppression because the beginJavadocTree() was giving `UWF_FIELD_NOT_INITIALIZED_IN_CONSTRUCTOR` error

Reply:
Why can’t we initialize fields?

Reply:
The error was saying that `private DetailNode parent;` is not initialized and dereferenced in `visitJavadocToken()` or something like that, I don't exactly remember it but it was something about not initializing `parent` variable in constructor and dereferencing it. 

In our case, beginJavadocTree() works like kind of a constructor, it initializes the `parent` variable before the visitJavadocTree() gets called.

Look, you will find this comment above these suppressions. It explains how the fields are initialized:
`<!-- false positive, beginTree is a kind of constructor for Checks -->`

@nrmancuso

Reply:
>Why can’t we initialize fields?

This is still not answered. Why can't we initialize this field?

Reply:
to initialize this field, we need the AST of the starting block of javadoc, we get this AST in the beginJavadocTree() and there we initialize this field. We cannot initialize this field where it is declared as we don't have access to the root AST. 

In our case beginJavadocTree() works as a constructor to initialize this field. 

The error was saying that we didn't initialized the `parent` field in constructor and we're dereferencing it in visitJavadocToken() but this is kind of false positive because the begin Tree method works as a constructor to initialize fields before the visit Token method is called. 

The field is actually initialized but not in a constructor but in the begin tree method. Which acts as a constructor for visit token method. That's why I had to add this suppression. I hope this clears your doubt. @nrmancuso

Reply:
>to initialize this field, we need the AST of the starting block of javadoc

Why can't we do `private DetailNode parent = null;`?

Reply:
> Why can't we do `private DetailNode parent = null;`?

I guess yes we can do this, this didn't cross my mind. I'll try this and push the changes

Reply:
We can keep this suppression, I agree.